compiler/grainformat/reformat.re

```diff
@@ -847,7 +856,14 @@ and print_constant = (c: Parsetree.constant) => {
     | PConstNumber(PConstNumberRational(n, d)) =>
       Printf.sprintf("%s/%s", n, d)
     | PConstBytes(b) => Printf.sprintf("%s", b)
-    | PConstChar(c) => Printf.sprintf("'%s'", String.escaped(c))
+    | PConstChar(c) =>
+      // String.escaped doesn't escape the single quote but we need it to
```

Comment:
Can you turn `String.escaped` into our own helper (as `Grain_utils.String_utils.escaped`) that does this logic? Any time we encounter ocaml stdlib functionality that we need to behave differently, we try to create a Grain_utils helper.

Reply:
I think my comment here is wrong.  A single quote doesn't need to be escaped in a string.  This is a very special case when printing a char.  What do you think?

Reply:
(comment updated)

Reply:
That's a good point. But do we use `String.escaped` anywhere else? Our parser wouldn't handle strings that weren't escaped correctly, if I'm not mistaken.

Reply:
The only other place is when formatting a string which is when it works properly, as an escaped \\' isn't legal

print("\'")
compiles to
Error: Illegal string character: "\\"

Reply:
Sounds good

Reply:
Oh, there's a Char.escaped!

Reply:
Fixed by using the proper function!